tests/unit/s2n_client_cert_verify_test.c

```diff
@@ -81,5 +136,55 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_cert_chain_and_key_free(chain_and_key));
     }
 
+    /*  Test: async private key operations. */
+    {
+        struct s2n_cert_chain_and_key *chain_and_key;
+        EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+                S2N_DEFAULT_TEST_CERT_CHAIN, S2N_DEFAULT_TEST_PRIVATE_KEY));
+
+        struct s2n_config *config;
+        EXPECT_NOT_NULL(config = s2n_config_new());
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key));
+        EXPECT_SUCCESS(s2n_config_set_async_pkey_callback(config, s2n_async_pkey_store_op));
```

Comment:
Is this async logic being used for both?

How can we actually be sure it is being used for both as it only has one set of variables which are shared by both the client and server?

Reply:
Yes, the async logic is being used for both, but the purpose of this test is really for the client.

To ensure the client is used for the async callback, we could do one of the following:
1. Split the config into a client and server, where only the client defines an async callback.
2. Count how many times the callback is invoked, and make sure both the client and server invoked it.

Reply:
I've updated the logic to only do async for the client connection